src/sql/SqlService.ts

```diff
@@ -0,0 +1,290 @@
+                'Client is not currently connected to the cluster.'
+            );
+        }
+
+        const queryId = SqlQueryId.fromMemberId(connection.getRemoteUuid());
+
+        const expectedResultType: SqlExpectedResultType =
+            sqlStatement.options?.expectedResultType ? SqlExpectedResultType[sqlStatement.options.expectedResultType]
+                : SqlServiceImpl.DEFAULT_EXPECTED_RESULT_TYPE;
+
+        try {
+            const serializedParams = [];
+            if (Array.isArray(sqlStatement.params)) { // params can be undefined
+                for (let param of sqlStatement.params) {
+                    param = this.convertToStringIfDatetimeValue(param);
+                    serializedParams.push(this.serializationService.toData(param));
+                }
+            }
+            const cursorBufferSize = sqlStatement.options?.cursorBufferSize ?
+                sqlStatement.options.cursorBufferSize : SqlServiceImpl.DEFAULT_CURSOR_BUFFER_SIZE;
+            const requestMessage = SqlExecuteCodec.encodeRequest(
+                sqlStatement.sql,
+                serializedParams,
+                sqlStatement.options?.timeoutMillis ? sqlStatement.options.timeoutMillis : SqlServiceImpl.DEFAULT_TIMEOUT,
+                cursorBufferSize,
+                sqlStatement.options?.schema ? sqlStatement.options.schema : SqlServiceImpl.DEFAULT_SCHEMA,
+                expectedResultType,
+                queryId
+            );
+
+            const res = SqlResultImpl.newResult(
+                this,
+                this.serializationService,
+                connection,
+                queryId,
+                cursorBufferSize,
+                sqlStatement.options?.hasOwnProperty('returnRawResult') ?
+                    sqlStatement.options.returnRawResult : SqlServiceImpl.DEFAULT_FOR_RETURN_RAW_RESULT,
+                this.connectionManager.getClientUuid()
+            );
+
+            this.invocationService.invokeOnConnection(connection, requestMessage).then(clientMessage => {
+                this.handleExecuteResponse(clientMessage, res);
+            }).catch(err => {
+                console.log(err);
+                res.onExecuteError(
+                    new HazelcastSqlException(
+                        connection.getRemoteUuid(), SqlErrorCode.CONNECTION_PROBLEM, err.message, err
+                    )
+                );
+            });
+
+            return res;
+        } catch (error) {
+            throw new HazelcastSqlException(
+                connection.getRemoteUuid(),
+                SqlErrorCode.GENERIC,
+                `An error occurred during SQL execution: ${error.message}`,
+                error
+            );
+        }
+    }
+
+    close(connection: Connection, queryId: SqlQueryId): Promise<void> {
+        const requestMessage = SqlCloseCodec.encodeRequest(queryId);
+        return this.invocationService.invokeOnConnection(connection, requestMessage).then(() => {
```

Comment:
Since this is just used by SqlResult internally, we can make this return directly the `return this.invocationService.invokeOnConnection(connection, requestMessage)`, instead of adding a no-op continuation